webapp/channels/src/components/spinner_button.tsx

```diff
@@ -1,37 +1,35 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {PureComponent} from 'react';
-import type {ButtonHTMLAttributes, ReactNode} from 'react';
+import React from 'react';
+import type {ReactNode} from 'react';
 
 import LoadingWrapper from 'components/widgets/loading/loading_wrapper';
 
-type Props = {
+type Props = React.ButtonHTMLAttributes<HTMLButtonElement> & {
     children?: ReactNode;
     spinning: boolean;
     spinningText: ReactNode;
 }
 
-export default class SpinnerButton extends PureComponent<Props & ButtonHTMLAttributes<HTMLButtonElement>> {
-    public static defaultProps: Partial<Props> = {
-        spinning: false,
-    };
+const SpinnerButton = ({spinning, spinningText, children, ...props}: Props) => {
+    return (
+        <button
+            disabled={spinning}
+            {...props}
+        >
+            <LoadingWrapper
+                loading={spinning}
+                text={spinningText}
+            >
+                {children}
+            </LoadingWrapper>
+        </button>
+    );
+};
 
-    public render(): JSX.Element {
-        const {spinning, spinningText, children, ...props} = this.props;
+SpinnerButton.defaultProps = {
```

Comment:
The usage of defaultProps in functional components is not recomended, if prepfered the default parameters, in this case, something like `const SpinnerButton = ({spinning = false, spinningText, children, ...props}: Props) => {`